### PR TITLE
docs(datepicker): add hide on scroll demo

### LINKED ISSUE #3709: docs(datepicker): add demo hide on scroll
hide datepicker when we scroll (#3340)

### PR DESCRIPTION
**Pull request by tarusin**
closes #3709

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+datepicker/demos/hide-on-scroll/hide-on-scroll.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  templateUrl: './hide-on-scroll.html'
+})
+export class DemoDatepickerHideOnScrollComponent {
+  @ViewChild(BsDatepickerDirective) datepicker: BsDatepickerDirective;
+
+  myForm = new FormGroup({
+    myDate: new FormControl(new Date())
+  });
+
+  @HostListener('window:scroll', ['$event'])
```
Comment (IlyaSurmay) [inline demo/src/app/components/+datepicker/demos/hide-on-scroll/hide-on-scroll.ts]: $event in redundant

File: demo/src/app/components/+datepicker/datepicker-section.list.ts [hunk 2]
```diff
@@ -63,6 +64,16 @@ export const demoComponentContent: ContentSection[] = [
         `,
         outlet: DemoDatePickerCustomFormatComponent
       },
+      {
+        title: 'Hide on scroll',
+        anchor: 'hide-on-scroll',
+        component: require('!!raw-loader?lang=typescript!./demos/hide-on-scroll/hide-on-scroll.ts'),
+        html: require('!!raw-loader?lang=markup!./demos/hide-on-scroll/hide-on-scroll.html'),
+        description: `
+          <p>Hide datepicker when scroll.</p>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+datepicker/datepicker-section.list.ts]: Hide the datepicker on page scroll.

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Got this error after unit tests run
unittesterror

**Review by EvilAlexei — CHANGES_REQUESTED** [review]
No link to original issue (closes/fixes/related to)

**Comment by YevheniiaMazur** [comment]
@tarusin , Please, fix conflicting files

**Review by IlyaSurmay — DISMISSED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by valorkin** [comment]
@tarusin remove from demo sample all code and html which doesn't relate to demo

**Review by valorkin — APPROVED** [review]

**Pull request merged**